File: src/domain/BookStatus.ts
export enum BookStatus {
  NotAdded = -1,
  Dropped = 0,
  Reading = 1,
  OnHold = 2,
  Finished = 3,
  PlanToRead = 4,
}

export default function parseBookStatus(status: BookStatus) {
  switch (status) {
    case BookStatus.Dropped:
      return "Desistido";
    case BookStatus.NotAdded:
      return "Não adicionado";
    case BookStatus.Reading:
      return "Lendo";
    case BookStatus.OnHold:
      return "Em espera";
    case BookStatus.Finished:
      return "Lido";
    case BookStatus.PlanToRead:
      return "Planejo ler";
    default:
      return "Status desconhecido";
  }
}

File: shelfe-android/domain/src/main/kotlin/com/androidvip/domain/BookStatus.kt
package com.androidvip.domain

enum class BookStatus(val value: Int) {
    NOT_ADDED(-1),
    DROPPED(0),
    READING(1),
    ON_HOLD(2),
    FINISHED(3),
    PLAN_TO_READ(4);

    companion object {
        operator fun get(value: Int): BookStatus {
            return values().find {
                it.value == value
            } ?: NOT_ADDED
        }
    }
}
